pkg/secrets/s3.go

```diff
@@ -0,0 +1,122 @@
+package secrets
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	yamlParse "gopkg.in/yaml.v2"
+	"reflect"
+	"strings"
+)
+
+type S3Secret struct {
+	region   string
+	bucket   string
+	filepath string
+	key      string
+}
+
+type S3Decrypter struct {
+	secretConfig string
+}
+
+func NewS3Decrypter(secretConfig string) *S3Decrypter {
+	return &S3Decrypter{secretConfig}
+}
+
+func (s3 *S3Decrypter) Decrypt() ([]byte, error) {
+	s3Secret, err := parseS3SecretConfig(s3.secretConfig)
+	if err != nil {
+		return []byte{}, err
+	}
+	secret, err := s3Secret.fetchSecret()
+	if err != nil {
+		return []byte{}, err
+	}
+
+	return secret, nil
+}
+
+func parseS3SecretConfig(secretConfig string) (S3Secret, error) {
+	if !strings.HasPrefix(secretConfig, "encrypted:") {
```

Comment:
I think this condition is already caught as long as you call the interface method. I’m not sure this is necessary

Reply:
Yeah, it's not necessary, I was just being extra careful. I'll take it out.